gems/pending/spec/middleware/events/middleware_event_monitor_spec.rb

```diff
@@ -0,0 +1,48 @@
+require "middleware/events/middleware_event_monitor"
+
+describe MiddlewareEventMonitor do
+  subject do
+    allow(MiqServer).to receive(:my_zone).and_return("default")
+    auth = AuthToken.new(:name => "test", :auth_key => "valid-token", :userid => "jdoe", :password => "password")
+    ems  = FactoryGirl.create(:ems_hawkular,
+                              :hostname        => 'localhost',
+                              :port            => 8080,
+                              :authentications => [auth])
+    WebMock.allow_net_connect!
+    described_class.new(ems)
+  end
+
+  context "#perf_catch_events" do
+    # VCR.eject_cassette
+    # VCR.turn_off!(ignore_cassettes: true)
+
+    VCR.configure do |c|
+      c.default_cassette_options = {
+        :match_requests_on => [:method, VCR.request_matchers.uri_without_params(:startTime, :endTime)]
```

Comment:
The cassette has the necessary urls but the urls have timestamp query params. The urls generated are different on each run because the timestamps are based on system time, and change.  This allows the urls to still match the cassette by telling VCR it does not need to match the timestamps exactly.